### PR TITLE
9303 test upgrading send-anywhere

### PR DESCRIPTION
**Pull request by turadg**
refs: #9303

## Description
This adds a test for upgrading send-anywhere. It doesn't yet pass without these in-flight PRs,
- https://github.com/Agoric/agoric-sdk/pull/9736
- https://github.com/Agoric/agoric-sdk/pull/9785

We've agreed to land this without those to reduce work-in-flight.

### Security Considerations
none

### Scaling Considerations
none

### Documentation Considerations
none

### Testing Considerations
The failing test has a link to the issue to make it pass.

### Upgrade Considerations
none

### CODE REVIEW

**Comment by turadg** [comment]
Experimenting with these merged:
- https://github.com/Agoric/agoric-sdk/pull/9736
- https://github.com/Agoric/agoric-sdk/pull/9785

File: packages/builders/scripts/testing/start-buggy-sendAnywhere.js [hunk 1]
```diff
@@ -64,18 +64,30 @@ export const startSendAnywhere = async ({
     }),
   );
 
+  const agoricNamesHangs = Far('agoricNames that hangs', {
+    // ...privateArgs.agoricNames,
+    lookup: async () => {
+      trace('agoricNames.lookup being called that will never resolve');
+      // BUG: this never resolves
+      return new Promise(() => {});
```
Comment (mhofman) [inline packages/builders/scripts/testing/start-buggy-sendAnywhere.js:73]: Can we add a comment of what exactly will hang in the test? I tracked down that `chainHub` uses `agoricNames` with `retriable`, so I assume something in the test will trigger usage of chainHub and its retriable vow to not be resolved before upgrade, but I lost track of how exactly that happens.

Also we should provide a little more details as to why this is a valid test (and equivalent to the previous one, which had the pending promise explicitly created in the vat being upgraded). Here the buggy `agoricNames` is external to the vat being upgraded, which means upgrading the vat will not reject the result promise for the `E(agoricNames).lookup()` call. However that call is simply awaited in a `retriable` async function, which means there is a local promise created and watched (the result of the async function), which is pending at vat upgrade, and which will be rejected on upgrade.

Interestingly, the lookup result promise will remain pending forever, causing a "leak" of the promise subscription (in the real world until the decider of the promise upgrades) even though the subscriber is actually no longer interested in the promise. This is a liveslots "limitation" that I created an issue about (#10101)

File: packages/orchestration/src/examples/send-anywhere.contract.js [hunk 2]
```diff
@@ -43,10 +46,16 @@ const contract = async (
 
   const creatorFacet = prepareChainHubAdmin(zone, chainHub);
 
+  // UNTIL https://github.com/Agoric/agoric-sdk/issues/9066
+  const logNode = E(privateArgs.storageNode).makeChildNode('log');
+  /** @type {(msg: string) => Vow<void>} */
+  const log = msg => vowTools.watch(E(logNode).setValue(msg));
```
Comment (mhofman) [inline packages/orchestration/src/examples/send-anywhere.contract.js:52]: It's expected that the logs of all invocations will write to the same node?

Comment (turadg) [reply packages/orchestration/src/examples/send-anywhere.contract.js:52]: Until #9066, yes.

File: packages/boot/test/orchestration/contract-upgrade.test.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  t.context = await makeWalletFactoryContext(
+    t,
+    '@agoric/vm-config/decentral-itest-orchestration-config.json',
+  );
+});
+test.after.always(t => t.context.shutdown?.());
+
+/**
+ * This test core-evals a buggy installation of the sendAnywhere contract by
+ * giving it a faulty `agoricNames` service with a lookup() function returns a
```
Comment (mhofman) [inline packages/boot/test/orchestration/contract-upgrade.test.ts]:
```suggestion
 * giving it a faulty `agoricNames` service with a lookup() function which returns a
```

File: packages/boot/test/orchestration/contract-upgrade.test.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+ * promise that never resolves.
+ *
+ * Because the send-anywhere flow requires a lookup(), it waits forever. This
+ * gives us a point at which we can upgrade the vat with a working agoricNames
+ * and see that the flow continues from that point.
+ */
```
Comment (mhofman) [inline packages/boot/test/orchestration/contract-upgrade.test.ts]: Can we clarify that the `lookup` call is not made directly in a flow, but instead from a host API which uses the `retriable` helper. As such it tests both the idempotent retry mechanism of `retriable` on upgrades, and the ability to resume an async-flow for which a host vow settles after an upgrade.

**Review by mhofman — APPROVED** [review]
LGTM, assuming some details are added on how the test works exactly. As I originally mentioned in https://github.com/Agoric/agoric-sdk/pull/9719/files#r1763727074, it took me some time to understand what was going on in this test.

**Pull request merged**